common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -436,11 +440,27 @@ public static ClientException clientExceptionFromException(@NonNull final Throwa
 
 
         if (e instanceof OutOfMemoryError) {
-            return new ClientException(
-                    ClientException.OUT_OF_MEMORY,
-                    e.getMessage(),
-                    e
-            );
+            // When receiving an out of memory error, instead of gracefully returning a failure result, we should shut down
+            // current broker process, to allow a new broker process to be launched at the next request from client app.
+            // This will result in a new broker process with fresh memory allocation, rather than repeating out of memory
+            // errors. In the case that this is an msal-only scenario, and this code is running inside client app, then
+            // client app will be shut down and will need to be launched again.
+
+            // Log status code and record exception in span
+            final Span currentSpan = SpanExtension.current();
+            currentSpan.setStatus(StatusCode.ERROR);
```

Comment:
Do these attributes actually get set? Asking because we are killing the process without calling span.end

Reply:
Added currentSpan.end()

Reply:
Ok, I would say that we should check in sandbox db if it is flowing correctly by explicitly throwing an exception.

Reply:
Resolving as i'm no longer crashing broker